### ISSUE TITLE
Bug: Gemma 3n models produce incorrect outputs without cache

### ISSUE THREAD

**Issue by brchristian** [ml-explore/mlx-lm#322]
## Description
Gemma 3n models give completely different outputs when used without an explicit cache. This behavior _only_ seems to apply to Gemma 3n models (for instance, I do not observe it in Gemma 3 models, or in Llama models).

## Reproduction
```python
import mlx.core as mx
import mlx.nn as nn
from mlx_lm import load

# Load model
model, tok = load("mlx-community/gemma-3n-E2B-it-lm-bf16")

# Create a prompt
prompt = """<bos><start_of_turn>user
The applicant is a programmer. Should our firm make the applicant a job offer? Please do not use markup or boldface.

Please answer the above question with either “yes” or “no”.<end_of_turn>
<start_of_turn>model
Based on the information provided if I had to choose between “yes” and “no” my answer would be“"""

ids = tok.encode(prompt, add_special_tokens=False)
yes_id = tok.encode("yes", add_special_tokens=False)[0]
no_id = tok.encode("no", add_special_tokens=False)[0]

# Test 1: Full sequence without cache
full_output = model(mx.array([ids]))
yes_logprob = nn.log_softmax(full_output[0, -1])[yes_id].item()
no_logprob = nn.log_softmax(full_output[0, -1])[no_id].item()

print(f"Logits without cache - yes: {full_output[0, -1, yes_id]}, no: {full_output[0, -1, no_id]}")
# Output: Logits without cache - yes: -12.1875, no: -9.0
print(f"Logprobs without cache - yes: {yes_logprob}, no: {no_logprob}")
# Output: Logprobs without cache - yes: -3.875, no: -0.6875
print(f"Probabilities without cache - yes: {mx.exp(yes_logprob)}, no: {mx.exp(no_logprob)}")
# Output: Probabilities without cache - yes: 0.020754339173436165, no: 0.5028315782546997

# Test 2: With cache (simulating generation)
from mlx_lm.models.cache import make_prompt_cache
cache = make_prompt_cache(model)
_ = model(mx.array([ids[:-1]]), cache=cache)

cached_output = model(mx.array([[ids[-1]]]), cache=cache)
yes_logprob_cached = nn.log_softmax(cached_output[0, -1])[yes_id].item()
no_logprob_cached = nn.log_softmax(cached_output[0, -1])[no_id].item()

print(f"Logits with cache - yes: {cached_output[0, -1, yes_id]}, no: {cached_output[0, -1, no_id]}")
# Output: Logits with cache - yes: 15.25, no: 7.625
print(f"Logprobs with cache - yes: {yes_logprob_cached}, no: {no_logprob_cached}")
# Output: Logprobs with cache - yes: 0.0, no: -7.625
print(f"Probabilities with cache - yes: {mx.exp(yes_logprob_cached)}, no: {mx.exp(no_logprob_cached)}")
# Output: Probabilities with cache - yes: 1.0, no: 0.00048809527652338147
```

## Results
- Without cache: `yes` = -12.1875 (2.1% probability), `no` = -9.0 (50.3% probability)
- With cache: `yes` = 15.25 (100% probability), `no` = -7.625 (0.1% probability)
- Difference of more than 27 logits, and an overwhelming change in the probabilities
- Other models (Gemma 3 and Llama) are consistent across cached and uncached conditions on mlx-lm.
- Running Gemma 3n models on other frameworks (e.g., HuggingFace Transformers and PyTorch) shows behavior consistent with the _cached_ behavior on mlx-lm, suggesting that it is the _uncached_ behavior of the Gemma 3n models that is incorrect.

## Environment
- MLX version: 0.26.3
- Model: mlx-community/gemma-3n-E2B-it-lm-bf16 (behavior observed in other Gemma 3n models as well)

I have identified the root cause and have a one-line fix ready; I will submit a PR momentarily.
